src/Compilers/Core/Portable/CodeGen/PrivateImplementationDetails.cs

```diff
@@ -210,10 +210,14 @@ internal Cci.IFieldReference CreateDataField(ImmutableArray<byte> data, ushort a
                 ((uint)data.Length, Alignment: alignment), key =>
                 key.Size switch
                 {
+                    // We need a type with the same size as the data.  If the size of the data is
+                    // 2, 4, or 8 bytes, we can use short, int, or long rather than creating a custom
+                    // type, but we can only do so if the required alignment is also 1, as these types
+                    // have a .pack value of 1.
                     1 when _systemInt8Type is not null => _systemInt8Type,
-                    2 when _systemInt16Type is not null => _systemInt16Type,
-                    4 when _systemInt32Type is not null => _systemInt32Type,
-                    8 when _systemInt64Type is not null => _systemInt64Type,
+                    2 when key.Alignment == 1 && _systemInt16Type is not null => _systemInt16Type,
```

Comment:
>key.Alignment == 1 && [](http://example.com/codeflow?start=27&length=21)

Consider refactoring the code to avoid duplicating this condition syntactically.
Also, is there a good reason to add these existing types into the ```_proxyTypes```? If not, we could avoid extra filtering in ```GetNestedTypes```. and could make type of ```_proxyTypes``` "stronger".

Reply:
> Consider refactoring the code to avoid duplicating this condition syntactically.

Ok

> Also, is there a good reason to add these existing types into the _proxyTypes?

I'm not sure; looks like it's been this way since the beginning:
https://github.com/dotnet/roslyn/blob/211b47ed044aaefca630a61d0d4bfcff192198ec/Src/Compilers/Core/Source/CodeGen/PrivateImplementationDetails.cs#L104
I'd like to avoid changing that as part of this, but it could certainly be investigated subsequently.